subsys/bluetooth/host/gatt.c

```diff
@@ -3564,6 +3599,26 @@ void bt_gatt_connected(struct bt_conn *conn)
 	data.conn = conn;
 	data.sec = BT_SECURITY_L1;
 
+#if IS_ENABLED(CONFIG_BT_SETTINGS_CCC_LOAD_ON_DEMAND)
+	/* Load CCC settings from backend if bonded */
+	if (bt_addr_le_is_bonded(conn->id, &conn->le.dst)) {
```

Comment:
Can't the IS_ENABLED() here be embedded into the if-statement? I.e.
```
if (IS_ENABLED(..) && bt_addr_le_is_bonded(..)) {
```

Reply:
Not easily without moving things around (it references ccc_set_direct with references ccc_set which is inside an #if defined), see also above answer to @Vudentz 's comment.

Reply:
Addressed in latest push, I separated functionality and moving functions around in separate commits as suggested by @joerchan.